main_comment.go

```diff
@@ -147,6 +151,46 @@ func processGitHubComment(
 	return nil
 }
 
+func protectBranch(conf *config, branchName string, pipelinePath string) error {
+	// https://docs.gitlab.com/ee/api/protected_branches.html#protect-repository-branches
+	opt := &gitlab.ProtectRepositoryBranchesOptions{
+		Name: &branchName,
+	}
+
+	client, err := clientgitlab.NewGitLabClient(
+		conf.gitlabToken,
+		conf.gitlabBaseURL,
+		conf.dryRunMode,
+	)
+
+	if err != nil {
+		return err
+	}
+
+	_, err = client.ProtectRepositoryBranches(pipelinePath, opt)
+	if err != nil {
+		return fmt.Errorf("%v returned error: %s", err, err.Error())
+	}
+	return nil
+}
+
+func syncProtectedBranch(
+	log *logrus.Entry,
+	pr *github.PullRequestEvent,
+	conf *config,
+	pipelinePath string,
+) (string, error) {
+	prBranchName := "pr_" + strconv.Itoa(pr.GetNumber()) + "_protected"
```

Comment:
We will not allow force-pushing, so by reusing the same branch name it won't be able to sync a second time. Let me explain:
* PR 123 launched with commit A
* command "start pipeline" -> the bot creates pr_123_protected and runs pipeline
* PR rebased or commit amended. Commit A'
* command "start pipeline" -> the bot tries to update pr_123_protected but GitLab force-push so it will fail

One solution is to delete the branch in advance if it exists. Another solution is to add a timestamp suffix into the branch name.

Reply:
I decided to approve the PR. We can follow-up on this on a separate PR once we have tested the basic functionality (so that we don't add complexity for the first try)